### PR TITLE
fix(snippets): trim leading whitespace when extracting snippets

### PR DESCRIPTION
**Pull request by chingor13**
When parsing snippets from the source file, trim leading spaces while maintaining extra indentation. This implementation finds the minimum indentation, ignoring empty lines and removes that many spaces from each line.

### CODE REVIEW

File: synthtool/gcp/snippets.py [hunk 1]
```diff
@@ -21,6 +21,24 @@
 CLOSE_SNIPPET_REGEX = r".*\[END ([a-z0-9_]+)\].*$"
 
 
+def _trim_leading_whitespace(lines: List[str]) -> List[str]:
+    def number_of_leading_spaces(input: str) -> int:
```
Comment (SurferJeffAtGoogle) [inline synthtool/gcp/snippets.py]: Please add a doc string.

Comment (chingor13) [reply synthtool/gcp/snippets.py]: Done.

**Comment by chingor13** [comment]
This will prevent PRs like https://github.com/googleapis/java-automl/pull/132

**Comment by chingor13** [comment]
@bcoe Is this going to cause an issue with your nodejs snippet usage?

**Pull request merged**